### DISCUSSION TITLE
Disable first element to be selected in the Select component

### DISCUSSION THREAD

**Discussion by brampurnot** [UI5/webcomponents-react#6229] [Q&A]
Is there a way to disable the selection of the first element in the Select component? I can always do some kind of hack but I assume there must be an easier way to do that.

**Comment by Lukas742** [comment]
Hi @brampurnot 

disabling options in a select is not supported by UX/A11y guidelines:

https://experience.sap.com/fiori-design-web/select/#option-list1
> Do not [disable](https://experience.sap.com/fiori-design-web/ui-element-states/#disabled) values. In most cases, it is completely unclear how to enable them. [Hide](https://experience.sap.com/fiori-design-web/ui-element-states/#hidden) the values instead.

**Reply by brampurnot** [reply]
Ok thanks! Is the onChange event also disabled on purpose or might this be a bug? It does not seem to be executed when it auto-selects the first option.

**Reply by Lukas742** [reply]
In most cases, components don't fire events for the initial state or if a state is updated programatically, as developers are then aware which state is applied. Only if a user changes the state, or if the internal state changes (e.g. validation), events are fired.

The same goes for native elements, e.g. `<select>` and `<option>` ([StackBlitz](https://stackblitz.com/edit/github-9tlhvb?file=src%2FApp.tsx)), so this behavior is intended.
